rules/alerting.go

```diff
@@ -113,7 +113,7 @@ type AlertingRule struct {
 	// The name of the alert.
 	name string
 	// The vector expression from which to generate alerts.
-	vector parser.Expr
+	vector fmt.Stringer
```

Comment:
This seems like an odd choice of type, why not a string? Why does this need to change at all? 

We've also had numerous bugs in the past from working with promql as rendered string rather than a parsed expression in recording rules, so we should not go back to that.

Reply:
This could definitely be `string`! I chose `Stringer` as it was the smallest code change that illustrated what I wanted. 

As for why it needs to change, it allows downstream consumers to use different types here that can be coercible to a string. I'm particularly interested in parsing LogQL queries in Loki in order to alert based upon logs.

Reply:
This still looks incorrect. Why does this need to change at all?